src/context.rs

```diff
@@ -1464,9 +1464,9 @@ impl BlockContext {
 
   pub fn set_cdef(&mut self, sbo: &SuperBlockOffset, cdef_index: u8) {
     let bo = sbo.block_offset(0, 0);
-    let bw = cmp::min (bo.x + 8, self.blocks[bo.y as usize].len());
-    let bh = cmp::min (bo.y + 8, self.blocks.len());
-
+    // Checkme: Is 16 still the right block unit for 128x128 superblocks?
+    let bw = cmp::min (bo.x + MAX_MIB_SIZE, self.blocks[bo.y as usize].len());
+    let bh = cmp::min (bo.y + MAX_MIB_SIZE, self.blocks.len());
     for y in bo.y..bh {
```

Comment:
Have you considered using for_each here?

Reply:
can't; if the superblock is bigger than the frame it will overflow the edge and panic.

Reply:
Sorry, I mean, if the superblock overlaps the edge of the frame.

Reply:
I see, I guess this relates to what we have discussed while ago.
I understand that, unless for_each has boundary check itself, you cant use it.

Reply:
[beta AWCY run incoming, should be fun....]

Reply:
ycho: exactly

Reply:
Lots of AWCY fails, only partial results.  Failed encodes succeed here, even with debugging builds. Hrm.